tests/simulator_test.cpp

```diff
@@ -2,82 +2,160 @@
 #include "main/simulator.h"
 #include "gtest/gtest.h"
 
-// ! OLD ENERGY COLLECTORS TEST
-/*
-TEST_F(SceneManagerTest, EnergyCollectorPositionsCheck) {
-  for (const auto &collectorPtr : manager.getEnergyCollectors()) {
-    std::cout << *collectorPtr << std::endl;
+using constants::kPi;
+using core::Ray;
+using core::RayHitData;
+using core::Vec3;
+
+const int kSkipNumCollectors = 37;
+const float kSkipFrequency = 1000;
+const Vec3 kVecZero(0, 0, 0);
+
+class FakeModel : public AbstractModel {
+
+public:
+  explicit FakeModel(bool empty) : empty_(empty){};
+  std::vector<objects::TriangleObj *> triangles() const override { return {}; }
+  float height() const override { return 0; }
+  float sideSize() const override { return 0; }
+  bool empty() const { return empty_; }
+
+private:
+  bool empty_;
+};
+
+class EnergyCollectorTest : public ::testing::Test {
+public:
+  EnergyCollectorTest() : nonEmptyModel(false), emptyModel(true){};
+  enum class HitResult { ENERGY_COLLECTORS_EMPTY, NO_HIT, HIT };
+
+protected:
+  std::vector<std::unique_ptr<objects::EnergyCollector>> energyCollectors;
+  FakeModel nonEmptyModel, emptyModel;
+
+  // performs ray hit at energy Collectors by modifying hitData.
+  // Returns:
+  // ENERGY_COLLECTORS_EMPTY - when no energyCollectors where assigned
+  // NO_HIT - when there was no hit
+  // HIT - when hit occurred;
+  [[nodiscard]] HitResult performHitCollector(const Ray &ray, float frequency,
+                                              RayHitData *hitData) {
+    if (energyCollectors.empty()) {
+      return HitResult::ENERGY_COLLECTORS_EMPTY;
+    }
+
+    for (const auto &collector : energyCollectors) {
+      if (collector->hitObject(ray, kSkipFrequency, hitData)) {
+        return HitResult::HIT;
+      }
+    }
+
+    return HitResult::NO_HIT;
   }
 
-  ASSERT_EQ(manager.getEnergyCollectors().size(), kPopulation);
+  void TearDown() override {
+    if (energyCollectors.empty()) {
```

Comment:
I read more carefully google test tear down and I deleted it becuase its not usefull in this case ;)

Reply:
What made you convinced that this is not needed? What about calling clear empty() before clear(), do you realize why it is not needed?

Reply:
clear() in vector calls destructor on each element, thats why empty() is not needed ;)

Reply:
What about the answer to question 1? What made you convinced that this is not needed? 

> clear() in vector calls destructor on each element, thats why empty() is not needed ;)

Even if clear() was not calling destructors, empty() would not be needed. So that is not the reason.